### PR TITLE
OCPCLOUD-2010: Re-vendor api and library-go for external platform support

### PR DESCRIPTION
**Pull request by adriengentil**
Revendor openshift/api and openshift/library-go in order to support the
external platform type.

library-go PR: https://github.com/openshift/library-go/pull/1496
API PR: https://github.com/openshift/api/pull/1434

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -120,3 +120,5 @@ require (
 	sigs.k8s.io/structured-merge-diff/v4 v4.2.3 // indirect
 	sigs.k8s.io/yaml v1.3.0 // indirect
 )
+
+replace vbom.ml/util => github.com/fvbommel/util v0.0.0-20180919145318-efcd4e0f9787
```
Comment (elmiko) [inline go.mod:124]: what's the story with this replacement?

**Review by elmiko — COMMENTED** [review]
just a question, otherwise lgtm

File: go.mod (hunk 1 above)
Comment (adriengentil) [reply go.mod:124]: Without it we get:
```
$ go mod tidy 
go: downloading vbom.ml/util v0.0.0-20180919145318-efcd4e0f9787
github.com/openshift/cluster-kube-controller-manager-operator/cmd/cluster-kube-controller-manager-operator imports
	github.com/openshift/library-go/pkg/operator/staticpod/prune tested by
	github.com/openshift/library-go/pkg/operator/staticpod/prune.test imports
	vbom.ml/util/sortorder: unrecognized import path "vbom.ml/util": https fetch: Get "https://vbom.ml/util?go-get=1": dial tcp: lookup vbom.ml: no such host
```
This library is an dependency of `go-library`.

It looks like this library was [moved to github](https://github.com/kubernetes/kubectl/issues/925) some time ago, and I saw this replacement in few [Openshift components](https://github.com/search?q=org%3Aopenshift%20%22replace%20vbom.ml%2Futil%22&type=code).

I guess the vendoring hid the issue up until now?

Comment (elmiko) [reply go.mod:124]: yeah, i guess so, thanks for the explanation!

Comment (ingvagabund) [reply go.mod:124]: Should not it be pulled from the library-go go.mod file automatically?

Comment (ingvagabund) [reply go.mod:124]: Running `go mod tidy` locally just changing api/library-go to `release-4.14` branch works without the need to set the replace.

Comment (ingvagabund) [reply go.mod:124]: https://github.com/openshift/cluster-kube-controller-manager-operator/compare/master...ingvagabund:cluster-kube-controller-manager-operator:go-mod-tidy-experiment?expand=1

Comment (adriengentil) [reply go.mod:124]: hum... which version of go do you use?
```
$ go version
go version go1.20.4 linux/amd64
```

Comment (ingvagabund) [reply go.mod:124]:
```
$ go version
go version go1.20.2 linux/amd64
```

Comment (ingvagabund) [reply go.mod:124]: The same works for me with 1.20.4 as well

Comment (adriengentil) [reply go.mod:124]: same it works for me with:
```
podman run --rm -v "$PWD":/usr/src/myapp -w /usr/src/myapp golang:1.20.4 sh -c "go mod tidy && go mod vendor && go mod verify"
```

but the one I have on my system (fedora) keeps failing :thinking:

Comment (adriengentil) [reply go.mod:124]: I can reproduce with the fedora image:
```
podman run --rm -v "$PWD":/usr/src/myapp -w /usr/src/myapp fedora sh -c "dnf install -y golang && go mod tidy && go mod vendor && go mod verify"
```

Comment (adriengentil) [reply go.mod]: Done!

**Pull request merged**